controllers/virtualmachine/virtualmachine/virtualmachine_controller.go

```diff
@@ -167,6 +178,54 @@ func classToVMMapperFn(
 	}
 }
 
+// encryptionClassToVMMapperFn returns a mapper function that can be used to
+// enqueue reconcile requests for VMs in response to an event on the
+// EncryptionClass resource.
+func encryptionClassToVMMapperFn(
+	ctx *pkgctx.ControllerManagerContext,
+	c client.Client) func(_ context.Context, o client.Object) []reconcile.Request {
+
+	// For a given EncryptionClass, return reconcile requests for VMs that
+	// specify the same EncryptionClass.
+	return func(_ context.Context, o client.Object) []reconcile.Request {
+		obj := o.(*byokv1.EncryptionClass)
+		logger := ctx.Logger.WithValues("name", obj.Name, "namespace", obj.Namespace)
+
+		logger.V(4).Info("Reconciling all VMs referencing an EncryptionClass")
+
+		// Find all VM resources that reference this EncryptionClass.
+		vmList := &vmopv1.VirtualMachineList{}
+		if err := c.List(ctx, vmList, client.InNamespace(obj.Namespace)); err != nil {
+			logger.Error(
+				err,
+				"Failed to list VirtualMachines for reconciliation due to EncryptionClass watch")
+			return nil
+		}
+
+		// Populate reconcile requests for VMs that reference this
+		// EncryptionClass.
+		var requests []reconcile.Request
+		for _, vm := range vmList.Items {
+			if vm.Spec.Crypto.EncryptionClassName == obj.Name {
+				requests = append(
+					requests,
+					reconcile.Request{
+						NamespacedName: client.ObjectKey{
+							Namespace: vm.Namespace,
+							Name:      vm.Name,
+						},
+					})
+			}
+		}
+
+		logger.Info(
```

Comment:
Also use `logger.V(4)` here to avoid log spam (since this will be called regardless of any requests exist)?